src/Bindings/ManualBindings.cpp

```diff
@@ -3116,30 +3114,57 @@ static int tolua_cLineBlockTracer_Trace(lua_State * tolua_S)
 	cLuaState L(tolua_S);
 	if (
 		!L.CheckParamUserType(idx, "cWorld") ||
-		!L.CheckParamTable   (idx + 1) ||
-		!L.CheckParamNumber  (idx + 2, idx + 7) ||
-		!L.CheckParamEnd     (idx + 8)
+		!L.CheckParamTable   (idx + 1)
 	)
 	{
 		return 0;
 	}
 
 	// Get the params:
 	cWorld * world;
-	double startX, startY, startZ;
-	double endX, endY, endZ;
+	Vector3d start;
+	Vector3d end;
 	cLuaState::cTableRefPtr callbacks;
-	if (!L.GetStackValues(idx, world, callbacks, startX, startY, startZ, endX, endY, endZ))
+	if (
+		L.CheckParamNumber  (idx + 2, idx + 7) &&
+		L.CheckParamEnd     (idx + 8)
+	)
+	{
+		if (!L.GetStackValues(idx, world, callbacks, start.x, start.y, start.z, end.x, end.y, end.z))
+		{
+			LOGWARNING("cLineBlockTracer:Trace(): Cannot read parameters (starting at idx %d), aborting the trace.", idx);
+			L.LogStackTrace();
+			L.LogStackValues("Values on the stack");
+			return 0;
+		}
+		else
+		{
+			LOGWARNING("cLineBlockTracer:Trace(): Deprecated overload should take two Vector3d.");
+		}
+
+	}
+	else if (
+		L.CheckParamNumber  (idx + 2, idx + 3) &&
+		L.CheckParamEnd     (idx + 4) &&
+		L.CheckParamUserType(idx + 2, "Vector3d") &&
+		L.CheckParamUserType(idx + 3, "Vector3d")
```

Comment:
As a question, xoft recently added transparent conversion between all the Vector3<> types for Lua, we may want `CheckParamVector3` here? Also for the `L.CheckParamNumber  (idx + 2, idx + 3)` above, surely we are expecting Vectors, not numbers?

Reply:
Yes of course I'm new to Lua, I used the wrong function. Now its changed.